mlflow/utils/requirements_utils.py

```diff
@@ -268,6 +269,27 @@ def _capture_imported_modules(model_uri, flavor):
 
 
 _MODULES_TO_PACKAGES = None
+_PACKAGES_TO_MODULES = None
+
+
+def _init_modules_and_packages_maps():
+    global _MODULES_TO_PACKAGES, _PACKAGES_TO_MODULES
+    if _MODULES_TO_PACKAGES is None and _PACKAGES_TO_MODULES is None:
+        # Note `importlib_metada.packages_distributions` only captures packages installed into
+        # Python’s site-packages directory via tools such as pip:
+        # https://importlib-metadata.readthedocs.io/en/latest/using.html#using-importlib-metadata
+        _MODULES_TO_PACKAGES = importlib_metadata.packages_distributions()
+
+        # In Databricks, `_MODULES_TO_PACKAGES` doesn't contain pyspark since it's not installed
+        # via pip or conda. To work around this issue, manually add pyspark.
+        if is_in_databricks_runtime():
+            _MODULES_TO_PACKAGES.update({"pyspark": ["pyspark"]})
+
+        _PACKAGES_TO_MODULES = {}
+        for module, pkg_list in _MODULES_TO_PACKAGES.items():
+            for pkg_name in pkg_list:
+                _PACKAGES_TO_MODULES[pkg_name] = module
```

Comment:
Can we separately initialize `_MODULES_TO_PACKAGES` and `_PACKAGES_TO_MODULES`?

Reply:
+1